File: ruby/src/Behavioral/Strategy/DvdNameContext.rb
#//DvdNameContext - the context

require "Behavioral/Strategy/DvdNameAllCapStrategy"
require "Behavioral/Strategy/DvdNameTheAtEndStrategy"
require "Behavioral/Strategy/DvdNameReplaceSpacesStrategy"

class DvdNameContext

	def initialize(strategyTypeIn)
 		@dvdNameStrategy = if strategyTypeIn == 'C'
 			DvdNameAllCapStrategy.new
		elsif strategyTypeIn == 'E'
 			DvdNameTheAtEndStrategy.new
		elsif strategyTypeIn == 'S'
 			DvdNameReplaceSpacesStrategy.new
		else
 			DvdNameTheAtEndStrategy.new
		end
	end

	def formatDvdNames ( names, replacementIn = ' ' )
		[names.map { |name| 
			@dvdNameStrategy.formatDvdName(name, replacementIn) }
		].join(",")
	end
end

File: scala/src/Behavioral/Strategy/DvdNameContext.scala
package Behavioral.Strategy;


//DvdNameContext - the context

public class DvdNameContext {
   private DvdNameStrategy dvdNameStrategy; 
   
   public DvdNameContext(char strategyTypeIn) {
       switch (strategyTypeIn) {
           case 'C' : 
             this.dvdNameStrategy = new DvdNameAllCapStrategy(); 
             break;
           case 'E' : 
             this.dvdNameStrategy = new DvdNameTheAtEndStrategy(); 
             break;
           case 'S' : 
             this.dvdNameStrategy = 
               new DvdNameReplaceSpacesStrategy();
             break;
           default  : 
             this.dvdNameStrategy = new DvdNameTheAtEndStrategy();
       }     
   }  
   
   public String[] formatDvdNames(String[] namesIn) {
       return this.formatDvdNames(namesIn, ' ');
   }    
   
   public String[] formatDvdNames(String[] namesIn, char replacementIn) {
       String[] namesOut = new String[namesIn.length];
       for (int i = 0; i < namesIn.length; i++) {
           namesOut[i] = 
             dvdNameStrategy.formatDvdName(namesIn[i], replacementIn);
       }
       return namesOut;
   }
}
